src/app/handlers/mod.rs

```diff
@@ -35,7 +35,7 @@ use tonic::transport::Channel;
 
 use crate::{
     app::{state::tabs::history::HistoryQuery, AppEvent},
-    ui::SelectedTab,
+    tui::SelectedTab,
```

Comment:
I renamed `ui` to `tui`

Reply:
i wont have time for a full review right now, but as a drive by comment i think what would align more to the structure i have in mind is renaming `app` to `tui`.  The `ui` code is actual display code for the `tui` app (i.e. it is `ratatui` widgets like `Table`, `Textarea`, `Paragraph`, etc.). i can imagine having a directory structure something like this:

tui/state
tui/handlers
tui/ui
cli/...
execution/...
...

Reply:
good call -- done in 9ea59d6